### PR TITLE
Feature: customizable shell command

### PR DESCRIPTION
**Pull request by xFA25E**
This is the first version of custamizable shell command with a list.

@pRot0ta1p @sirikid

### CODE REVIEW

File: sdcv.el [hunk 1]
```diff
@@ -253,19 +251,20 @@
...
-  :type 'list
+  :type '(repeat string)
   :group 'sdcv)
 
 (defcustom sdcv-dictionary-data-dir nil
   "Default, sdcv search word from /usr/share/startdict/dict/.
-You can customize this value with local dir,
-then you don't need copy dict data to /usr/share directory everytime when you finish system install."
-  :type 'string
+You can customize this value with local dir,then you don't need copy dict data
```
Comment (sirikid) [inline sdcv.el]: Missing space after comma

File: sdcv.el [hunk 2]
```diff
@@ -410,25 +445,27 @@ And show information use tooltip."
...
-(defun sdcv-previous-dictionary ()
-  "Jump to previous dictionary."
-  (interactive)
-  (show-all)
-  (if (search-backward-regexp "^-->.*\n-" nil t) ;don't show error when search failed
-      (progn
-        (forward-char 1)
-        (recenter 0))                   ;adjust position
-    (message "Reached first dictionary.")))
+(with-no-warnings
```
Comment (sirikid) [inline sdcv.el]: What kinds of warnings you want to suppress?

File: sdcv.el [hunk 3]
```diff
@@ -440,15 +477,16 @@ And show information use tooltip."
...
-      (when (looking-at outline-regexp)
-        (show-entry)))))
+(with-no-warnings
+  (defun sdcv-next-line (arg)
+	"Next ARG line and show item."
+	(interactive "P")
+	(ignore-errors
+	  (call-interactively 'next-line arg)
+	  (save-excursion
+		(beginning-of-line nil)
```
Comment (sirikid) [inline sdcv.el]: Tabs

File: sdcv.el [hunk 4]
```diff
@@ -593,39 +625,38 @@ Argument DICTIONARY-LIST the word that need transform."
...
-    (dolist (word words)
-      (if (and (>= char-offset search-index)
-               (<= char-offset (+ search-index (length word))))
-          (return word)
-        (setq search-index (+ search-index (length word))))
-      )))
+	(catch 'result
+	  (dolist (word words)
+		(if (and (>= char-offset search-index)
+				 (<= char-offset (+ search-index (length word))))
```
Comment (sirikid) [inline sdcv.el]: Tabs again

Comment (sirikid) [inline sdcv.el]: You can use single `<=` here

File: sdcv.el [hunk 5]
```diff
@@ -644,16 +675,17 @@ Argument SDCV-STRING the search string from sdcv."
...
+	"Re-initialize buffer.
 Hide all entry but the first one and goto
 the beginning of the buffer."
-  (ignore-errors
-    (setq buffer-read-only t)
-    (goto-char (point-min))
-    (sdcv-next-dictionary)
-    (show-all)
-    (message "Finished searching `%s'." sdcv-current-translate-object)))
+	(ignore-errors
```
Comment (sirikid) [inline sdcv.el]: Tabs

**Review by sirikid — CHANGES_REQUESTED** [review]

**Comment by xFA25E** [comment]
Thanks for the review!

**Review by sirikid — APPROVED** [review]

**Review by ghost — APPROVED** [review]
Looks good to me

**Comment by manateelazycat** [comment]
First, thanks for patch.

Please split the patch, each patch only do one thing, all the content is mixed together, I can't review.

I don't understand what is the purpose of custom shell command?

**Comment by xFA25E** [comment]
The purpose is to make sdcv shell command customizable

**Comment by xFA25E** [comment]
You are right, I should split the patch. I'll send a splitted version soon

**Pull request closed without merging**